App crash after flash a module

### Please check before submitting an issue/在提交 issue 前请检查

- [x] I have searched the issues and haven't found anything relevant/我已经搜索了 issues 列表，没有发现于本问题相关内容
- [x] If patch failed, root failed, or device unable to boot after flashing the new boot.img. Please goto KernelPatch/修复失败或刷入修补后镜像不能启动，请前往 KernelPatch 提问
- [x] I will upload bugreport file in APatch Manager - Settings - Report log/我会上传 bureport 文件从 APatch 管理器 - 设置 - 发送日志
- [x] I know how to reproduce the issue which may not be specific to my device/我知道如何重新复现这个问题

### Version requirement/版本要求

- [x] I am using latest CI version of APatch/我正在使用最新 CI 版本

### Describe the bug/描述 bug

After flash PUI Themed Icon Completion Project on Oneplus ACE2V, the Apatch stop for a while and show several chown error.
Similar with chown the monochrome.png:0
Then it suddenly show the crash page.
But later I can see the module installed after restart APatch.

### Reproduce method/复现方法

1. download the latest module file from PUI group
2. flash with latest Apatch
3. wait for complete
4. see some chown error
5. the app crash

### Expected behavior/预期行为

install the module without crash

### Actual behaviour /实际行为

crash when nearly installed the module

### Screenshots/截图

_No response_

### Logs/日志

APatch version: 11039 (11039)

Brand: OnePlus
Model: PHP110
SDK Level: 35
Time: 2025-02-14 23:04:18

Thread: main
Crash Info: 
java.lang.RuntimeException: android.os.TransactionTooLargeException: data parcel size 5066424 bytes
Bundle stats:
  androidx.lifecycle.BundlableSavedStateRegistry.key [size=5065432]
    lV:-1 [size=5064360]
      -ipdx3q [size=5058844]
      9uksvn [size=5460]
PersistableBundle stats:
  [null]
	at android.app.servertransaction.PendingTransactionActions$StopInfo.run(PendingTransactionActions.java:146)
	at android.os.Handler.handleCallback(Handler.java:959)
	at android.os.Handler.dispatchMessage(Handler.java:100)
	at android.os.Looper.loopOnce(Looper.java:282)
	at android.os.Looper.loop(Looper.java:387)
	at android.app.ActivityThread.main(ActivityThread.java:9505)
	at java.lang.reflect.Method.invoke(Native Method)
	at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:600)
	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:1005)
Caused by: android.os.TransactionTooLargeException: data parcel size 5066424 bytes
	at android.os.BinderProxy.transactNative(Native Method)
	at android.os.BinderProxy.transact(BinderProxy.java:679)
	at android.app.IActivityClientController$Stub$Proxy.activityStopped(IActivityClientController.java:1545)
	at android.app.ActivityClient.activityStopped(ActivityClient.java:120)
	at android.app.servertransaction.PendingTransactionActions$StopInfo.run(PendingTransactionActions.java:135)
	... 8 more

### Device Name/设备名称

Oneplus ACE2V

### OS Version/系统版本

ColorOS15 Android15

### APatch Version/APatch 版本

11039

### Kernel Version/内核版本

5.10.209-android12-9-o-gdbe4d00409d9

### KernelPatch Version/KernelPatch 版本

0.11.2

### Additional context/其他信息

[APatch_bugreport_2025-02-14_23_15.tar.gz](https://github.com/user-attachments/files/18801150/APatch_bugreport_2025-02-14_23_15.tar.gz)

--- a/app/src/main/java/me/bmax/apatch/ui/screen/ExecuteAPMAction.kt
+++ b/app/src/main/java/me/bmax/apatch/ui/screen/ExecuteAPMAction.kt
@@ -21,6 +21,7 @@ import androidx.compose.runtime.Composable
 import androidx.compose.runtime.LaunchedEffect
 import androidx.compose.runtime.getValue
 import androidx.compose.runtime.mutableStateOf
+import androidx.compose.runtime.remember
 import androidx.compose.runtime.rememberCoroutineScope
 import androidx.compose.runtime.saveable.rememberSaveable
 import androidx.compose.runtime.setValue
@@ -50,12 +51,17 @@ import java.util.Locale
 @Destination<RootGraph>
 fun ExecuteAPMActionScreen(navigator: DestinationsNavigator, moduleId: String) {
     var text by rememberSaveable { mutableStateOf("") }
-    var tempText: String
-    val logContent = rememberSaveable { StringBuilder() }
+    val logContent = remember { StringBuilder() }
     val snackBarHost = LocalSnackbarHost.current
     val scope = rememberCoroutineScope()
     val scrollState = rememberScrollState()
     var actionResult: Boolean
+
+    fun appendLog(line: String) {
+        logContent.append(line).append("\n")
+        val newText = text + line + "\n"
+        text = if (newText.length > 100_000) newText.takeLast(100_000) else newText
+    }
 
     LaunchedEffect(Unit) {
         if (text.isNotEmpty()) {
@@ -65,16 +71,14 @@ fun ExecuteAPMActionScreen(navigator: DestinationsNavigator, moduleId: String) {
             runAPModuleAction(
                 moduleId,
                 onStdout = {
-                    tempText = "$it\n"
-                    if (tempText.startsWith("[H[J")) { // clear command
-                        text = tempText.substring(6)
+                    if (it.startsWith("\u001B[H\u001B[2J")) { // clear command
+                        text = it.substring(9)
                     } else {
-                        text += tempText
+                        appendLog(it)
                     }
-                    logContent.append(it).append("\n")
                 },
                 onStderr = {
-                    logContent.append(it).append("\n")
+                    appendLog(it)
                 }
             ).let {
                 actionResult = it

--- a/app/src/main/java/me/bmax/apatch/ui/screen/Install.kt
+++ b/app/src/main/java/me/bmax/apatch/ui/screen/Install.kt
@@ -76,10 +76,15 @@ enum class MODULE_TYPE {
 @Composable
 @Destination<RootGraph>
 fun InstallScreen(navigator: DestinationsNavigator, uri: Uri, type: MODULE_TYPE) {
-    var text by remember { mutableStateOf("") }
-    var tempText: String
+    var text by rememberSaveable { mutableStateOf("") }
     val logContent = remember { StringBuilder() }
     var showFloatAction by rememberSaveable { mutableStateOf(false) }
+
+    fun appendLog(line: String) {
+        logContent.append(line).append("\n")
+        val newText = text + line + "\n"
+        text = if (newText.length > 100_000) newText.takeLast(100_000) else newText
+    }
     val metaModuleAlertDialog = rememberCustomDialog { dismiss: () -> Unit ->
         val uriHandler = LocalUriHandler.current
         AlertDialog(
@@ -141,21 +146,17 @@ fun InstallScreen(navigator: DestinationsNavigator, uri: Uri, type: MODULE_TYPE)
                 }
 
             }, onStdout = {
-                tempText = "$it\n"
-                if (tempText.startsWith("[H[J")) { // clear command
-                    text = tempText.substring(6)
+                if (it.startsWith("[H[J")) { // clear command
+                    text = it.substring(5)
                 } else {
-                    text += tempText
+                    appendLog(it)
                 }
-                logContent.append(it).append("\n")
             }, onStderr = {
-                tempText = "$it\n"
-                if (tempText.startsWith("[H[J")) { // clear command
-                    text = tempText.substring(6)
+                if (it.startsWith("[H[J")) { // clear command
+                    text = it.substring(5)
                 } else {
-                    text += tempText
+                    appendLog(it)
                 }
-                logContent.append(it).append("\n")
             })
         }
     }

